src/dashboard/apps/core/models.py

```diff
@@ -58,19 +63,22 @@ class DeliveryPoint(DashboardBase):
     """Represents a delivery point for electric vehicles.
 
     Attributes:
-    - provider_id (CharField): stores the unique identifier for the delivery provider.
-    - entities (ManyToManyField): linking DeliveryPoint to associated Entity instances.
+    - delivery_point_id (CharField): stores the unique identifier for
+    the delivery point.
+    - entity (ForeignKey): linking DeliveryPoint to associated Entity.
     - is_active (BooleanField): indicating the active status of the delivery point.
     """
 
-    provider_id = models.CharField(_("provider id"), max_length=64)
-    entities = models.ManyToManyField(Entity, verbose_name=_("entities"))
+    delivery_point_id = models.CharField(_("delivery point id"), max_length=64)
```

Comment:
As discussed, the following might be less confusing:

```suggestion
    provider_assigned_id = models.CharField(_("delivery point id"), max_length=64)
```